src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs

```diff
@@ -94,55 +98,27 @@ internal partial class NavigationBarController
         }
     }
 
-    /// <summary>
-    /// Starts a new task to compute what item should be selected.
-    /// </summary>
-    private void StartSelectedItemUpdateTask()
-    {
-        // Cancel any in flight work.  This way we don't update until a short lull after the last user event we received.
-        _selectItemQueue.AddWork(cancelExistingWork: true);
-    }
-
-    private async ValueTask SelectItemAsync(CancellationToken cancellationToken)
+    private async ValueTask SelectItemAsync(ImmutableSegmentedList<int> positions, CancellationToken cancellationToken)
     {
-        // Switch to the UI so we can determine where the user is and determine the state the last time we updated
-        // the UI.
-        await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken).NoThrowAwaitable();
-
-        // Cancellation exceptions are ignored in AsyncBatchingWorkQueue, so return without throwing if cancellation
-        // occurred while switching to the main thread.
-        if (cancellationToken.IsCancellationRequested)
-            return;
+        var lastCaretPosition = positions.Last();
 
-        await SelectItemWorkerAsync(cancellationToken).ConfigureAwait(true);
+        // Can grab this directly here as only this queue ever reads or writes to it.
+        var lastPresentedInfo = _lastPresentedInfo;
 
-        // Once we've computed and selected the latest navbar items, pause ourselves if we're no longer visible.
-        // That way we don't consume any machine resources that the user won't even notice.
-        if (_visibilityTracker?.IsVisible(_subjectBuffer) is false)
-            Pause();
-    }
+        // Make a task that waits indefinitely, or until the cancellation token is signaled.
+        var cancellationTriggeredTask = Task.Delay(-1, cancellationToken);
```

Comment:
>Task.Delay(-1, cancellationToken); [](http://example.com/codeflow?start=40&length=34)

Weird, this seems like a common enough desire that I would have thought there would be something like a CancellationToken.ToTask() method or equivalent somewhere to hide the backing Task.Delay(-1, ...).